### PR TITLE
Updates and new questions

### PR DESCRIPTION
**Pull request by alain-lclark**
Peter,

I've made some changes, added some questions, and fixed a typo.  Feel free to use any or all parts of these changes.

Alain

### CODE REVIEW

File: data_structures/variables.md [hunk 1]
```diff
@@ -64,6 +64,24 @@ The *scope* of a variable is the region of code where the variable can be access
     int variable = 2;
     int VARIABLE = 3;
     ```
+1. :star: What is the scope of variable `n` in the following code snippet?
+   ```java
+   int fac(int n) {
+       int p = 1;
+       for (int i = 1; i <= n; i++)
+           p = p * i;
```
Comment (PeterDrake) [inline data_structures/variables.md:72]: I prefer to ALWAYS use curly braces with loops and if statements.

File: data_structures/variables.md [hunk 2]
```diff
@@ -64,6 +64,24 @@ The *scope* of a variable is the region of code where the variable can be access
...
+       return p;
+   }
+   ```
+1. :star: What is the scope of variable `i` in the following code snippet?
+   ```java
+   int fac(int n) {
+       int p = 1;
+       for (int i = 1; i <= n; i++)
+           p = p * i;
```
Comment (PeterDrake) [inline data_structures/variables.md:81]: Same here.

File: development_tools/intellij_idea.md [hunk 3]
```diff
@@ -62,7 +61,11 @@ delicious gravy to be had. My general advice is to keep an eye out for anything
 a better way; often there is a way to get the IDE to do the boring work for you. Don't get hung up on memorizing all of the
 options, which would be impossible; just learn the features you use often and be vaguely aware of the features you use occasionally (so you can look them up when you need them).
 ### Creating a Program
-Right click on the `src` directory and select `New` | `Java Class`.
+Right click on the `src` directory, select `New` | `Java Class`, select `Class`, and choose a name. It's probably wise also
+to associate your program with a run configuration as follows. Click on the gray down arrow near the green "play" triangle
+near the top center of the window and select `Edit Configurations...`, click on the `+` button in the top left corner,
+select `Application`, give your configuration a `Name` (typically the same as the name of the class you've just chosen),
+type the name of your class again in the `Main class` field, and click `OK`.
```
Comment (PeterDrake) [inline development_tools/intellij_idea.md:68]: I never bother with this step. I believe a new configuration is created when you first run the file, and the default configuration works fine.

**Review by PeterDrake — COMMENTED** [review]
Thanks for these submissions! See my individual comments; after we get those resolved we can merge them in.

File: data_structures/variables.md (hunk 1 above)
Comment (alain-lclark) [reply data_structures/variables.md:72]: I'll fix that.

File: development_tools/intellij_idea.md (hunk 3 above)
Comment (alain-lclark) [reply development_tools/intellij_idea.md:68]: I thought I was running into issues if I didn't explicitly created configurations.  But I wasn't able to repeat these behaviors.  I didn't realize IntelliJ was creating them automatically when clicking on the "play" button near the class name.  I'll drop this change.

**Comment by PeterDrake** [comment]
Alain, did we want to continue with this pull request? It looks like there are unresolved issues.

File: data_structures/variables.md (hunk 1 above)
Comment (alain-lclark) [reply data_structures/variables.md:72]: I believe those changes are now in the main line.

File: data_structures/variables.md (hunk 2 above)
Comment (alain-lclark) [reply data_structures/variables.md:81]: Same here.

File: development_tools/intellij_idea.md (hunk 3 above)
Comment (alain-lclark) [reply development_tools/intellij_idea.md:68]: Based on your comments and my improved understanding of IntelliJ I've decided to abandoned these changes.

**Comment by alain-lclark** [comment]
I believe all the changes I cared about have made it to the main line.

I'm closing pull request.

**Pull request closed without merging**